relay/relay.go

```diff
@@ -5,6 +5,7 @@ import (
 	"context"
```

Comment:
Copy of my comment, as it's easier to discuss here:
> Have you tried it and made sure it works with e.g. DNS connection reset by peer errors like the ticket?

Reply:
It should work for any error related to establishment of the connection because I wrapped the entire `c.connect()` setup call. I didn't try specifically with DNS timeouts/connection refused (tried disconnecting local network, overriding API name with something incorrect) but it should work all the same.